06-building-a-program-start-to-finish/solving-the-problem/output/invertedIndex.js

```diff
@@ -0,0 +1,165 @@
+/**
+ * The Inverted Index class.
+ * @class
+ */
+class InvertedIndex {
+
+  /**
+   * class constructor
+   * @constructor
+   */
+  constructor() {
+    this.unique = [];
+    this.tableObj = {};
+  }
+
+  /**
+   * A method that tokenizes the string that is passed through it
+   * @param  {string} data the string  from the "text" key in the JSON
+   * @return {Array} containing String.
+   */
+  tokenize(data) {
+    this.terms = data.replace(/[^\w\s]/gi, ' ')
+      .match(/\w+/g);
+    return this.terms;
+  }
+
+  /**
+   * A method that filters the array passed into it for unique words
+   * @param {String} data Array of strings
+   * @returns {String} Returns array of unique words
+   */
+  uniqueWords(data) {
+    this.unique = [...new Set(data)];
+    return this.unique;
+  }
+
+  /**
+   * A method that tokenizes an object and gets the unique terms in the object
+   * @param  {Object} fileJson the uploaded JSON file object
+   * @return {Array}  of the unique terms in the fileJson
+   */
+  getTextFromJsonObj(fileJson) {
+    let newText = ' ';
+    Object.keys(fileJson)
+      .forEach((key) => {
+        const obj = fileJson[key];
+        newText += ' ';
+        newText += obj.text;
+      });
+    const uniqueTerms = this.uniqueWords(this.tokenize(newText))
+      .map(x => x.toLowerCase());
+    return uniqueTerms;
+  }
+
+  /**
+   * A method to validate a JSON file.
+   * it check if the format of the .json file contains
+   * strictly keys of ["title","text"]
+   * @param {Object} parsedFile object
+   * @returns {Object} containing boolean and a String.
+   */
+  validateFile(parsedFile) {
+    this.isValid = {
+      valid: false,
+      message: 'This is an Invalid JSON File',
+    };
+    Object.keys(parsedFile)
+      .forEach((key) => {
+        const uploadedFile = parsedFile[key];
+        const validFormat = ['title', 'text'];
+        const parsedFileFormat = Object.keys(uploadedFile);
+        const fileTextKey = Object.keys(parsedFileFormat)
+          .map(objKeys => parsedFileFormat[objKeys]);
+        if (validFormat.toString() === fileTextKey.toString()) {
+          this.isValid = {
+            valid: true,
+            message: 'This JSON Format is correct',
+          };
+        } else {
+          this.isValid = {
+            valid: false,
+            message: 'This JSON Format is Incorrect',
+          };
+        }
+      });
+    return this.isValid;
+  }
+
+  /**
+   * A method that create the index of the selected-file
+   * @param  {Object} fileJson the uploaded file object
+   * @param  {Array} uniqueTerms    the unique terms in the JSON file
+   * @param  {String} fileName    The file name
+   * @return {Object} containing the index of the selected-file
+   */
+  createIndex(fileJson, uniqueTerms, fileName) {
```

Comment:
If the `createIndex` function is supposed to create an index from the file passed why is it taking an array of `uniqueTerms`?

Should it do that as a process to creating the index?